### DISCUSSION TITLE
Converting Legacy PXE Boot to UEFI

### DISCUSSION THREAD

**Discussion by elikirk** [ipxe/ipxe#410] [Q&A]
Hi all! This may be a little long-winded but I want to make sure I'm verbose in describing my situation. Currently, I have a diskless laptop and a laptop that will be connected to an SSD. On this SSD I have installed the OS that I want my diskless client to boot to via iSCSI and iPXE. I have demonstrated success by having DHCP hand out undionly.kpxe via TFTP during a Legacy boot from Onboard NIC. In this case, the SSD is MBR-partitioned with an ext4 filesystem.

The new obstacle I have run into is that I have to perform a UEFI boot instead. So at this point I have configured by diskless laptop to UEFI boot over IPv4, disabled secure boot, and formatted the SSD to a GPT schema. I installed the OS to the disk and have been attempting to boot by initially handing out ipxe.efi that is built from source and has the intel, iscsi, and debug flags built into it. 

Currently, this schema gets into the iPXE shell and tries to sanboot, but fails with two errors:

1. EFIBLK 0x80 could not load drivers https://ipxe.org/err/7f22208e
2. EFIBLK 0x80 could not locate a filesystem https://ipxe.org/err/7f22208e

At this point I'm a little confused as to what else I need to do to finish this "conversion". For reference, I am testing with a base RHEL 8.4 install. My iSCSI targets make use of fileio objects, so the SSD is mounted at a directory /home/pxe which simply contains a .img file. Any and all help is greatly appreciated!

**Comment by NiKiZe** [comment]
For EFI boot the bootx64.efi needs to be located at the correct location on a fat32 partition.

First make a disk that you can boot from in EFI mode locally, then make that disk available over iSCSI, then modify and compare.
